pkg/aws/permissions.go

```diff
@@ -80,93 +77,3 @@ func (c *awsClient) ValidateSCP(target *string, policies map[string]string) (boo
 
 	return true, nil
 }
-
-// checkPermissionsUsingQueryClient will use queryClient to query whether the credentials in targetClient can perform
-// the actions listed in the statementEntries. queryClient will need
-// sts:GetCallerIdentity and iam:SimulatePrincipalPolicy
-func checkPermissionsUsingQueryClient(queryClient *awsClient, targetUserARN string, policyDocument PolicyDocument,
-	params *SimulateParams) (bool, error) {
-	// Ignoring isRoot here since we only warn the user that its not best practice to use it.
-	// TODO: Add a check for isRoot in the initialize
-	allowList := []*string{}
-	for _, statement := range policyDocument.Statement {
-		actionArr := getActionAllowed(statement.Action)
-		for _, action := range actionArr {
-			allowList = append(allowList, aws.String(action))
-		}
-	}
-
-	input := &iam.SimulatePrincipalPolicyInput{
-		PolicySourceArn: aws.String(targetUserARN),
-		ActionNames:     allowList,
-		ContextEntries:  []*iam.ContextEntry{},
-	}
-
-	if params != nil {
-		if params.Region != "" {
-			input.ContextEntries = append(input.ContextEntries, &iam.ContextEntry{
-				ContextKeyName:   aws.String("aws:RequestedRegion"),
-				ContextKeyType:   aws.String("stringList"),
-				ContextKeyValues: []*string{aws.String(params.Region)},
-			})
-		}
-	}
-
-	// Either all actions are allowed and we'll return 'true', or it's a failure
-	allClear := true
```

Comment:
No need for this flag, we can just check the length of the `failedActions` instead.